### PR TITLE
Mark `IsolatedProject` as _unsupported_ by the Configuration Cache 

### PR DESCRIPTION
**Pull request by bamboo**
Follow-up to #28011 .

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by alllex — APPROVED** [review]
LGTM!

File: platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheUnsupportedTypesIntegrationTest.groovy [hunk 1]
```diff
@@ -17,6 +17,7 @@
 package org.gradle.configurationcache
 
 import org.gradle.api.DefaultTask
+import org.gradle.api.IsolatedProject
```
Comment (alllex) [inline platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheUnsupportedTypesIntegrationTest.groovy:20]: 🤔 Looking at the import I realized we placed the new type in a very crowded package.

Wouldn't it be better to choose a more confined space for it? Some ideas:
- `org.gradle.api.project`
- `org.gradle.api.isolated`

**Comment by bamboo** [comment]
@bot-gradle test ACC

File: platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheUnsupportedTypesIntegrationTest.groovy (hunk 1 above)
Comment (bamboo) [reply platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheUnsupportedTypesIntegrationTest.groovy:20]: Good point.

**Comment by bamboo** [comment]
@bot-gradle merge

**Pull request merged**